Exercise 3 Recursive Unhider/recursiveUnhider-rt/src/com/tridiumuniversity/recursiveUnhider/BRecursiveUnhideJob.java

```diff
@@ -0,0 +1,111 @@
+package com.tridiumuniversity.recursiveUnhider;
+
+import javax.baja.job.BSimpleJob;
+import javax.baja.job.JobCancelException;
+import javax.baja.naming.BOrd;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+
+public class BRecursiveUnhideJob extends BSimpleJob
+{
+/*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+/*@ $com.tridiumuniversity.recursiveUnhider.BRecursiveUnhideJob(2979906276)1.0$ @*/
+/* Generated Mon Sep 25 13:09:22 CEST 2023 by Slot-o-Matic (c) Tridium, Inc. 2012 */
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+  
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BRecursiveUnhideJob.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+    @Override
+    public void run(Context context)
+    {
+        // Find BRecursiveUnhiderService
+        BRecursiveUnhiderService recursiveUnhiderService = (BRecursiveUnhiderService) Sys.getService(BRecursiveUnhiderService.TYPE);
+        if (recursiveUnhiderService == null)
+        {
+            throw new IllegalStateException("Recursive Unhider Service not available.");
+        }
+
+        log().start("Preparing to search for hidden components...");
+
+        // Prepare query to find hidden components
+        BOrd rootOrd = recursiveUnhiderService.getRootOrd();
+        BComponent rootComponent = (BComponent) rootOrd.get();
```

Comment:
In real production code, it'd be a good idea to wrap this in a try/catch block to handle the `UnresolvedException` if the ORD doesn't exist. If you really want to play it safe, you could also check that `rootOrd` resolves to a `BComponent` before casting it.